presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -1574,11 +1574,17 @@ public static long widthBucket(@SqlType(StandardTypes.DOUBLE) double operand, @S
         double bin;
 
         while (lower < upper) {
+            if (bins.isNull(lower) || bins.isNull(upper - 1)) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Bin values cannot be NULL");
+            }
             if (DOUBLE.getDouble(bins, lower) > DOUBLE.getDouble(bins, upper - 1)) {
                 throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Bin values are not sorted in ascending order");
             }
 
             index = (lower + upper) / 2;
+            if (bins.isNull(index)) {
```

Comment:
the null checks showing up 2x makes me wonder if we should also be checking isFinite on lower/upper and checking sortedness for index (lower < index < upper).

Reply:
We can do that.
Do we want it?
Then the scope of this change will grow past just nulls.

Reply:
probably we do want it, but we can leave this change to just nulls.

Reply:
oh never mind, saw you updated already. Thanks!